bundles/org.eclipse.e4.core.commands/src/org/eclipse/e4/core/commands/internal/HandlerServiceHandler.java

```diff
@@ -33,13 +35,15 @@
 /**
  * Provide an IHandler to delegate calls to.
  */
-public class HandlerServiceHandler extends AbstractHandler {
+public class HandlerServiceHandler extends AbstractHandlerWithState {
 
 	private static final String FAILED_TO_FIND_HANDLER_DURING_EXECUTION = "Failed to find handler during execution"; //$NON-NLS-1$
 	private static final String HANDLER_MISSING_EXECUTE_ANNOTATION = " handler is missing @Execute"; //$NON-NLS-1$
 	private static final Object missingExecute = new Object();
 
 	protected final String commandId;
+	// Remove state from currentStateHandler when it goes out of scope
+	private IObjectWithState currentStateHandler;
```

Comment:
This field leaks whole editor imstances, see new 4 test fails:

https://download.eclipse.org/eclipse/downloads/drops4/I20231107-1800/testresults/html/org.eclipse.jdt.ui.tests_ep430I-unit-cen64-gtk3-java17_linux.gtk.x86_64_17.html

Please provide fix ASAP.

Reply:
Quick fix is impossible, as the field was an important part of the original PR. We have to revert the whole thing.
@mickaelistria could you please do so?

Reply:
If its really a leak, why not use a `WeakReference` here?

Reply:
@laeubi yes, weak reference might work. I will attempt to apply it now.

Reply:
Have no IDE at hand, but isn't possible to clear the reference on dispose()?

Reply:
@iloveeclipse AFAIK, most of these handlers have lifetime of Workbench.

Reply:
Created https://github.com/eclipse-platform/eclipse.platform.ui/pull/1291 with `WeakReference`. There is a problem with it (objects are collected without removal of listeners), but it does not leak.
The test for leaks would be hard to write.